### PR TITLE
Dragonflight - Evoker Setup

### PR DESCRIPTION
**Pull request by danny-janse**
Probably merging this into the wrong branch as this is actually meant for dragonflight but i cant find a way to make branches on my own. I'm happy to re-sumbit this PR if a dragonflight branch can be made.

Screenshot 2022-07-22 at 09 17 28

This PR adds support for both Evoker specializations excluding actual parsers and spells, but including;
* Dragonflight expansion entry
* Evoker Class
* Devastation Evoker
* Perservation Evoker
* Power Type: Essence (This is exlusive to Evoker)
* Both Alliance and Horde Race: Dracthyr
* Class and Specs icons

Changes nothing to do with the new Evoker class; 
* Change to monk color; This was to close to the Evoker and could cause confusion
* Updated Balance Druid mastery spell
* Fix to warlock translation (wouldn't load wowanalyzer locally without this change)
* Fix to SpecList styling for incomplete rows. (2 warrior specs would take 100% rather than 66.7%)

### CODE REVIEW

**Comment by danny-janse** [comment]
I have several other branches open that build upon this PR. Would you like me to add them onto this PR?

[djanse/dragonflight-evoker-spells](https://github.com/djanse/WoWAnalyzer/tree/dragonflight-evoker-spells)
[djanse/evoker-blessing-bronze](https://github.com/djanse/WoWAnalyzer/tree/evoker-blessing-bronze)

**Comment by tyndivelspaz** [comment]
Beat me to the punch on making a PR. I just finished with the talents and abilities and spells for preservation and started on devastation. I'll hold off doing much more and merge my changes with these

**Comment by abelito75** [comment]
This shouldn't go into the shadowlands branch

File: src/common/SPELLS/talents/evoker.ts
Comment (abelito75) [inline src/common/SPELLS/talents/evoker.ts]: feral?

File: src/interface/report/CombatantInfoFaker.ts
Comment (abelito75) [inline src/interface/report/CombatantInfoFaker.ts:140]: I'm not 100% but I'm pretty certain that this whole class can and probably should be deleted and it was hyper specific to shadowlands beta
@Pewtro Should know since he was the creator of it

**Review by abelito75 — CHANGES_REQUESTED** [review]
This still shouldn't be in the shadowlands branch. @emallson or another admin (idk if I can do it or not) should create a df-prepatch branch where we stage this...
There probably needs to be more thought put into branches for this since if people proactively update the shadowlands branch when df-prepatch also exists we might have a lot of FUN conflicts in the future... 

Maybe we should just hold off on the creation of this till DF-prepatch as most alpha/beta things are highly subject to change

File: src/common/SPELLS/talents/evoker.ts
Comment (danny-janse) [reply src/common/SPELLS/talents/evoker.ts]: Ah yes! i compied the balance file. I updated this in another ready to go PR that is based on this where it has been renamed. But Looking over the changes @Darthelit made on the file i think its best to accept their PR on the SpellList.

Has been updated.

File: src/interface/report/CombatantInfoFaker.ts
Comment (danny-janse) [reply src/interface/report/CombatantInfoFaker.ts:140]: - Adds new class (game class) 
- Gets confused about this comment

Took me a while to understand haha :)

**Comment by danny-janse** [comment]
> This still shouldn't be in the shadowlands branch. @emallson or another admin (idk if I can do it or not) should create a df-prepatch branch where we stage this... There probably needs to be more thought put into branches for this since if people proactively update the shadowlands branch when df-prepatch also exists we might have a lot of FUN conflicts in the future...
> 
> Maybe we should just hold off on the creation of this till DF-prepatch as most alpha/beta things are highly subject to change

i whole heartedly agree, hence the first thing i mentioned with this PR was the non-existent dragonflight branch. I just had some time today, added the class and submitted the PR so that another person wouldn't have to. I wasn't personally planning on changing/adding modules.

File: src/interface/report/CombatantInfoFaker.ts
Comment (Pewtro) [reply src/interface/report/CombatantInfoFaker.ts:140]: If combatantinfo isn’t available once scripts are enabled then it could be useful again, but for now probably no need to add it unless it starts breaking stuff

Comment (danny-janse) [reply src/interface/report/CombatantInfoFaker.ts:140]: Do I need to make changes to the CombatantnfoFaker?

Comment (Pewtro) [reply src/interface/report/CombatantInfoFaker.ts:140]: No, I’ll do a pass at it later in the cycle once scripts are enabled to see if it has merit or not.

Comment (emallson) [reply src/interface/report/CombatantInfoFaker.ts:140]: we currently have combatantinfo, but no talent info in there yet.

**Comment by emallson** [comment]
I don't want to create a `dragonflight` branch *quite* yet because we're not ready to start on core systems work and keeping the two branches up-to-date in parallel will be a pain. However, we can leave this PR open and merge it into the DF branch once we create it. The only thing that should end up with a conflict is the changelog

**Comment by danny-janse** [comment]
> I don't want to create a `dragonflight` branch _quite_ yet because we're not ready to start on core systems work and keeping the two branches up-to-date in parallel will be a pain. However, we can leave this PR open and merge it into the DF branch once we create it. The only thing that should end up with a conflict is the changelog

Sounds good. If it helps in the overview of PRs i can close the PR and you can hit me up anytime to re-open it?

**Comment by emallson** [comment]
Lets convert to draft instead---don't want somebody else duplicating this work since they missed the closed PR

**Comment by danny-janse** [comment]
i'll have to make some adjustmentds after #5061 has been merged. I'll follow the 'disabled by default' template in this.

**Comment by danny-janse** [comment]
Closing because this is now part of #5063

**Pull request closed without merging**
